language/stdlib/modules/DiemAccount.move

```diff
@@ -1650,27 +1652,38 @@ module DiemAccount {
                 balance_amount >= max_transaction_fee,
                 Errors::invalid_argument(PROLOGUE_ECANT_PAY_GAS_DEPOSIT)
             );
+            // [PCA8]: Check that the gas fee can be paid in this currency
+            assert(
+                TransactionFee::is_coin_initialized<Token>(),
+                Errors::invalid_argument(PROLOGUE_ECANT_PAY_GAS_DEPOSIT)
```

Comment:
IMO, this should really have a separate error code. This one translates to a validation status of `INSUFFICIENT_BALANCE_FOR_TRANSACTION_FEE`. But, this check has nothing to do with the account's balance, and it would be confusing to report that when an account has a sufficient balance of the specified gas currency. I will add a new error code for this.

Reply:
One other option here would be to have this translate out to an `INVALID_GAS_SPECIFIER` validation status. This seems  in line with the error (that the gas currency provided is not registered as a "gas currency") while not introducing a new validation status code.

Reply:
`INVALID_GAS_SPECIFIER` describes the case where the specifier is not a valid identifier (or perhaps with https://github.com/libra/libra/issues/6082 that it uses lowercase characters). I don't think it would be good to combine that with checking if the currency is registered for transaction fees.